src/account.ts

```diff
@@ -1,23 +1,113 @@
-import * as ethjsUtil from 'ethjs-util'
+import * as assert from 'assert'
+import * as BN from 'bn.js'
+import * as rlp from 'rlp'
+import { stripHexPrefix } from 'ethjs-util'
+import { KECCAK256_RLP, KECCAK256_NULL } from './constants'
+import { zeros, bufferToHex, toBuffer } from './bytes'
+import { keccak, keccak256, keccakFromString, rlphash } from './hash'
+import { assertIsHexString, assertIsBuffer } from './helpers'
+import { BNLike, BufferLike, bnToRlp } from './types'
+
 const {
   privateKeyVerify,
   publicKeyCreate,
   publicKeyVerify,
   publicKeyConvert,
 } = require('ethereum-cryptography/secp256k1')
-import * as assert from 'assert'
-import * as BN from 'bn.js'
-import { zeros, bufferToHex } from './bytes'
-import { keccak, keccak256, keccakFromString, rlphash } from './hash'
-import { assertIsHexString, assertIsBuffer } from './helpers'
 
-/**
- * Returns a zero address.
- */
-export const zeroAddress = function(): string {
-  const addressLength = 20
-  const addr = zeros(addressLength)
-  return bufferToHex(addr)
+export interface AccountData {
+  nonce?: BNLike
+  balance?: BNLike
+  stateRoot?: BufferLike
+  codeHash?: BufferLike
+}
+
+export class Account {
+  nonce: BN
+  balance: BN
+  stateRoot: Buffer
+  codeHash: Buffer
+
+  static fromAccountData(accountData: AccountData) {
+    const { nonce, balance, stateRoot, codeHash } = accountData
+
+    return new Account(
+      nonce ? new BN(toBuffer(nonce)) : undefined,
+      balance ? new BN(toBuffer(balance)) : undefined,
+      stateRoot ? toBuffer(stateRoot) : undefined,
+      codeHash ? toBuffer(codeHash) : undefined,
+    )
+  }
+
+  public static fromRlpSerializedAccount(serialized: Buffer) {
+    const values = rlp.decode(serialized)
+
+    if (!Array.isArray(values)) {
+      throw new Error('Invalid serialized account input. Must be array')
+    }
+
+    return this.fromValuesArray(values)
+  }
+
+  public static fromValuesArray(values: Buffer[]) {
+    const [nonce, balance, stateRoot, codeHash] = values
+
+    return new Account(
+      nonce ? new BN(nonce) : undefined,
+      balance ? new BN(balance) : undefined,
+      stateRoot,
+      codeHash,
+    )
+  }
+
+  /**
+   * This constructor takes the values, validates and assigns them.
+   * Use the static factory methods to assist in creating an Account from varying data types.
+   */
+  constructor(
+    nonce = new BN(0),
+    balance = new BN(0),
+    stateRoot = KECCAK256_RLP,
+    codeHash = KECCAK256_NULL,
+  ) {
+    if (stateRoot.length !== 32) {
+      throw new Error('stateRoot must have a length of 32')
+    }
+    if (codeHash.length !== 32) {
+      throw new Error('codeHash must have a length of 32')
+    }
+
```

Comment:
Do we want to add `nonce` and `balance` >= 0 checks here as well or is this unnecessary?

Reply:
sure, added in https://github.com/ethereumjs/ethereumjs-util/pull/275/commits/b0aacc0eff2ded594a12b614508a294ad9c63f4b